In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Why do people go to the DMV?
https://yogov.org/blog/most-popular-reasons-visit-dmv/

This data is from california.

| Percent | Reason                                                                       |
|---------|------------------------------------------------------------------------------|
| 11.9    | behind the wheel                                                             |
| 5.8     | transfer vechicle title                                                      |
| 15.3    | permit Test                                                                  |
| 25.5    | First license                                                                |
| 15.1    | Other (changing address, changing names, getting license plates, and others) |
| 13.7    | Renew License                                                                |
| 12.7    | Register Vehicle                                                             |

Condense

| Percent            | Reason                    | Actions needed |
|--------------------|---------------------------|----------------|
| 11.9               | Driving Test              | RT, Camera     |
| 5.8+15.1+12.7=33.6 | No Test                   | (no clerk)     |
| 25.5+15.3=40.8     | permit test/First license | WT             |
| 13.7               | Renew Licese              | Camera         |

## Parameter Estimates
p_checkin_written: .408

p_checkin_clerk: .336

p_checkin_camera: .137

p_checkin_driving: .119

# Rejection
Customers will be rejected after a written test, road test, check in or clerk.

$$\begin{aligned}
\Pr(\text{reject})
&= \Pr(\text{reject|wt}) \\
&= \Pr(\text{reject|rt}) \\
&= \Pr(\text{reject|ci}) \\
&= \Pr(\text{reject|clerk}) \\
\end{aligned}$$

From https://yogov.org/dmv/whats-the-most-common-reason-to-go-the-dmv/ Customers said that 33% of the time their issue was not resolved(rejected). https://www.dmv.ca.gov/portal/uploads/2020/04/S2-154.pdf Gives the fail rates for the tests.

let $a$ be the probability of not having the right paperwork at CI. $B$ Be the probability of not having the right paperwork at Clerk.

$$\begin{aligned}
\Pr(\text{reject})
&= .55(.408)(1-a) \\
&+ .456(.119)(1-a) \\
&+ a \\
&+ b(.336+.137+(1-.456)(.119))(1-a) \\
\end{aligned}$$

We other scenarios is the probability of not having the proper paper work. Usually the job of the ci is to weed out customers that dont have proper paper work ahead of time. We will say that ci catchs about 80% of these cases. Thus $a/(b+a)=.8 \rightarrow a=4b$.


$$\begin{aligned}
\frac{1}{3}
&= .55(.408)(1-a) \\
&+ .456(.119)(1-a) \\
&+ a \\
&+ .25a(.336+.137+(1-.456)(.119))(1-a) \\
\end{aligned}$$

Solve for a using wolf ram alpha
$$a \approx 0.0645375$$

## Updated Parameter Estimates
p_checkin_written: .408 (1-a)

p_checkin_clerk: .336 (1-a)

p_checkin_camera: .137 (1-a)

p_checkin_driving: .119 (1-a)

p_checkin_fail: a

---
p_written_fail: .55

p_road_fail: .456

p_clerk_fail: a/4

# How many people go to the DMV?
https://www.dmv.ca.gov/portal/dmv-research-reports/department-of-motor-vehicles-dmv-performance-reports/ gives us the total number of vechicles registered at the dmv. We can scale this to total number of arrivals. We also have the number of offices in 2022. https://www.dmv.ca.gov/portal/file/california-dmv-statistics-pdf/, which shouldnt be far off from the number in 2017

In [15]:
# Approx people per day at the dmv
bussiness_days_year = 260
success_rate = 2/3
prob_registering = .127
num_vechicles_registered = 4974226
num_dmvs = 173


total_arrivals = num_vechicles_registered/(success_rate*prob_registering)

total_arrivals = total_arrivals/bussiness_days_year # Per day
total_arrivals = total_arrivals/num_dmvs # Per dmv
# total_arrivals = total_arrivals / 9.5 # Per hour
total_arrivals

1306.1516404491233

We would like to have an average of around 1306 people visit the dmv per day. We also have 2 busy periods, one in the morning and one during lunch.

## Parameter Estimates
morning_rush_end = 0.5 # until 8:00

lunch_rush_start = 3.5 # 11:00 start

lunch_rush_end = 6.5 # 2:00 end

---

decreased_arrival_rate = 110

increased_arrival_rate = 160

In [14]:
weights = [1,.5,3,3,3]
rates = [50,162,115,162,115]
np.dot(weights,rates)

1307.0